astropy/io/ascii/cds.py

```diff
@@ -164,14 +203,493 @@ def get_cols(self, lines):
+                col.meta.unit = col.unit.to_string("cds")
+            elif col.name.lower().find("magnitude") > -1:
+                # ``col.unit`` can still be ``None``, if the unit of column values
+                # is ``Magnitude``, because ``astropy.units.Magnitude`` is actually a class.
+                # Unlike other units which are instances of ``astropy.units.Unit``,
+                # application of the ``Magnitude`` unit calculates the logarithm
+                # of the values. Thus, the only way to check for if the column values
+                # have ``Magnitude`` unit is to check the column name.
+                col.meta.unit = "mag"
+            else:
+                col.meta.unit = "---"
+
+            # Add col limit values to col description
+            lim_vals = ""
+            if (col.min and col.max and
+                not any(x in col.name for x in ['RA', 'DE', 'LON', 'LAT'])):
+                # No col limit values for coordinate columns.
+                if col.fortran_format[0] == 'I':
+                    if abs(col.min) < MAX_COL_INTLIMIT and abs(col.max) < MAX_COL_INTLIMIT:
+                        if col.min == col.max:
+                            lim_vals = "[{0}]".format(col.min)
+                        else:
+                            lim_vals = "[{0}/{1}]".format(col.min, col.max)
+                elif col.fortran_format[0] in ('E', 'F'):
+                    lim_vals = "[{0}/{1}]".format(math.floor(col.min * 100) / 100.,
+                                                  math.ceil(col.max * 100) / 100.)
+
+            if lim_vals != '' or nullflag != '':
+                description = "{0}{1} {2}".format(lim_vals, nullflag, description)
+
+            # Find the maximum label and description column widths.
+            if len(col.name) > max_label_width:
+                max_label_width = len(col.name)
+            if len(description) > max_descrip_size:
+                max_descrip_size = len(description)
+
+            # Add a row for the Sign of Declination in the bbb table
+            if col.name == 'DEd' and col[0] < 0.0:
+                bbb.add_row([singlebfmt.format(startb),
+                             "A1", "---", "DE-",
+                             "Sign of Declination"])
+                startb += 1
+
+            # Add Byte-By-Byte row to bbb table
+            bbb.add_row([singlebfmt.format(startb) if startb == endb
+                            else fmtb.format(startb, endb),
+                         "" if col.fortran_format is None else col.fortran_format,
+                         col.meta.unit,
+                         "" if col.name is None else col.name,
+                         description])
+            startb = endb + 2
+
+        # Properly format bbb columns
+        bbblines = StringIO()
+        bbb.write(bbblines, format='ascii.fixed_width_no_header',
+                  delimiter=' ', bookend=False, delimiter_pad=None,
+                  formats={'Format': '<6s',
```

Comment:
The `Format` and `Units` can both be longer than 6 characters, so I'm confused about the hardwired 6 here.

Reply:
Yeah, they can be. **6** is hardcoded to have a minimum column width when `Format` and `Units` have less than 6 characters. This also helps in the Byte-By-Byte mini-table header row alignment.